yb-voyager/cmd/exportDataStatus.go

```diff
@@ -152,13 +159,20 @@ func exportDataStatus(ctx context.Context, tablesProgressMetadata map[string]*ut
 			fmt.Println(ctx.Err())
 			break
 		}
+		// Wait for metadata to update and then exit (fix for empty partitions in MySQL, Oracle)
+		if updateMetadataAndExit && (source.DBType == MYSQL || source.DBType == ORACLE) {
```

Comment:
Get rid of the explicit check on the DBType. Have a consistent behaviour irrespective of source DB Type.